packages/crypto/src/secp256k1.rs

```diff
@@ -349,6 +367,32 @@ mod tests {
             let pubkey = secp256k1_recover_pubkey(&message_hash, &r_s, recovery_param).unwrap();
             assert_eq!(pubkey, expected);
         }
+
+        let file = File::open(COSMOS_SECP256K1_TESTS_JSON).unwrap();
+        let reader = BufReader::new(file);
+        let codes: Vec<Encoded> = serde_json::from_reader(reader).unwrap();
+        for (i, encoded) in (1..).zip(codes) {
+            let message = hex::decode(&encoded.message).unwrap();
+            let signature = hex::decode(&encoded.signature).unwrap();
+            let public_key = hex::decode(&encoded.public_key).unwrap();
+
+            let hash = hex::decode(&encoded.message_hash).unwrap();
+            let message_hash = Sha256::digest(message);
+            assert_eq!(hash.as_slice(), message_hash.as_slice());
+
+            // Since the recovery param is mossing in the test vectors, we try both 0 and 1
```

Comment:
```suggestion
            // Since the recovery param is missing in the test vectors, we try both 0 and 1
```

I understand that in a real scenario you need to know the recovery param. Otherwise, you can get a recovered pubkey that is not the right one.

I also understand that this is not related to the high/low-S comments, but another issue / improvement.

Will read more about all this and comment back if needed.

Reply:
> I understand that in a real scenario you need to know the recovery param. Otherwise, you can get a recovered pubkey that is not the right one.

> 
> Will read more about all this and comment back if needed.

Here are some references about the recovery param:

 - https://ethereum.stackexchange.com/questions/42455/during-ecdsa-signing-how-do-i-generate-the-recovery-id
 - https://bitcoin.stackexchange.com/questions/38351/ecdsa-v-r-s-what-is-v

Reply:
> I also understand that this is not related to the high/low-S comments, but another issue / improvement.

Exactly. The only reason they are mixed together here is because I pull in the COSMOS_SECP256K1_TESTS_JSON tests which are missing the recovery param.